File: 01. Jan_2025/01Jan/printSetsOfAnagrams.java
class Solution {
    public ArrayList<ArrayList<String>> anagrams(String[] arr) {
        Map<String, List<String>> map = new HashMap<>();        
        for (String str : arr) {
            char[] chars = str.toCharArray();
            Arrays.sort(chars);
            String sorted = new String(chars);
            if (!map.containsKey(sorted)) {
                map.put(sorted, new ArrayList<>());
            }
            map.get(sorted).add(str);
        }
        ArrayList<ArrayList<String>> res = new ArrayList<>();
        for (List<String> group : map.values()) {
            res.add(new ArrayList<>(group));
        }
        return res;
    }
}

File: 01. Jan_2025/01Jan/printSetsOfAnagrams.cpp
class Solution {
public:
    vector<vector<string>> anagrams(vector<string> &arr) {
        unordered_map<string, vector<string>> map;
        for (string str : arr) {
            string sorted = str;
            sort(sorted.begin(), sorted.end());
            map[sorted].push_back(str);
        }
        vector<vector<string>> res;
        for (auto &group : map) {
            res.push_back(group.second);
        }        
        return res;
    }
};